io.c

```diff
@@ -10922,62 +10918,82 @@ pipe_pair_close(VALUE rw)
 
 /*
  *  call-seq:
- *     IO.pipe                             ->  [read_io, write_io]
- *     IO.pipe(ext_enc)                    ->  [read_io, write_io]
- *     IO.pipe("ext_enc:int_enc" [, opt])  ->  [read_io, write_io]
- *     IO.pipe(ext_enc, int_enc [, opt])   ->  [read_io, write_io]
+ *    IO.pipe(enc_string = 'UTF-8', **opts) -> [read_io, write_io]
+ *    IO.pipe(enc_string = 'UTF-8', **opts) {|read_io, write_io] ...} -> object
+ *    IO.pipe(int_enc = Encoding::UTF_8, ext_enc = nil, **opts) -> [read_io, write_io]
+ *    IO.pipe(int_enc = Encoding::UTF_8, ext_enc = nil, **opts) {|read_io, write_io] ...} -> object
  *
- *     IO.pipe(...) {|read_io, write_io| ... }
+ *  Creates a pair of pipe endpoints, +read_io+ and +write_io+,
+ *  connected to each other.
  *
- *  Creates a pair of pipe endpoints (connected to each other) and
- *  returns them as a two-element array of IO objects:
- *  <code>[</code> <i>read_io</i>, <i>write_io</i> <code>]</code>.
+ *  If argument +enc_string+ is given, it must be a string containing one of:
  *
- *  If a block is given, the block is called and
- *  returns the value of the block.
- *  <i>read_io</i> and <i>write_io</i> are sent to the block as arguments.
- *  If read_io and write_io are not closed when the block exits, they are closed.
- *  i.e. closing read_io and/or write_io doesn't cause an error.
+ *  - The name of the encoding to be used as the internal encoding.
+ *  - The colon-separated names of two encodings to be used as the internal
+ *    and external encodings.
  *
- *  Not available on all platforms.
+ *  An array of the encoding names is returned by method Encoding.name_list.
```

Comment:
This paragraph was meant to be a part of the list above, right?

Reply:
I thought it best to describe the argument fully first, then tell where to find the names.